qcore/im.py

```diff
@@ -99,3 +117,45 @@ def order_ims(unsorted_ims, pattern_order=DEFAULT_PATTERN_ORDER):
         [adj_ims.append(im) for im in unsorted_ims if im not in adj_ims]
 
     return adj_ims
+
+
+@dataclass
+class IM:
+    name: [IMEnum]
```

Comment:
How come you have square brackets around the type?

Reply:
Great question... 😅